lib/elixir/lib/supervisor.ex

```diff
@@ -487,7 +487,9 @@ defmodule Supervisor do
   init callback to return the proper supervision flags.
   """
   @callback init(init_arg :: term) ::
-              {:ok, {:supervisor.sup_flags(), [:supervisor.child_spec()]}}
+              {:ok,
+               {sup_flags() | (old_erlang_sup_flags :: :supervisor.sup_flags()),
+                [child_spec() | (old_erlang_child_spec :: :supervisor.child_spec())]}}
```

Comment:
This is correct but your question makes me think this is a good time as any to actually deprecate the old style flags, so let's go with:

```suggestion
               {sup_flags(), [child_spec() | (old_erlang_child_spec :: :supervisor.child_spec())]}}
```

Reply:
Ok. Merged.